src/librustdoc/clean/mod.rs

```diff
@@ -579,8 +583,61 @@ fn is_elided_lifetime(param: &hir::GenericParam<'_>) -> bool {
 
 pub(crate) fn clean_generics<'tcx>(
     gens: &hir::Generics<'tcx>,
+    def_id: DefId,
     cx: &mut DocContext<'tcx>,
+    is_in_impl: bool,
 ) -> Generics {
+    let mut bounds_to_remove: FxHashMap<Type, (Vec<GenericBound>, Vec<Lifetime>)> =
+        Default::default();
+    let mut regions_to_remove: FxHashMap<Lifetime, Vec<GenericBound>> = Default::default();
+
+    // There can't be duplicates between the item's generics and its parent's if we're not in an
+    // impl block.
+    if is_in_impl {
+        // FIXME: Once we can differentiate between generics (`T: Foo`) and where predicates
+        // (`where T: Foo`), we should remove the `gens` argument and directly build from the
+        // results returned here.
+        let substs = InternalSubsts::identity_for_item(cx.tcx, def_id);
+        let predicates =
+            cx.tcx.explicit_predicates_of(def_id).instantiate(cx.tcx, substs).predicates;
+        let param_env = cx.tcx.param_env(cx.tcx.parent(def_id));
+        let param_env = EarlyBinder(param_env).subst(cx.tcx, substs);
+
+        for predicate in predicates {
+            let infcx = cx.tcx.infer_ctxt().build();
+            let obligation =
+                Obligation::new(infcx.tcx, ObligationCause::dummy(), param_env, predicate);
+            let obligation_ctxt = ObligationCtxt::new(&infcx);
```

Comment:
nit: `ocx`

Reply:
I guess I'll go for so it's coherent with the compiler once perf check is done.

Reply:
Done!